docusaurus/docs/Android/02-client/06-guides/01-push-notifications.md

```diff
@@ -2,72 +2,141 @@
-:::
+#### Step 1
+Configure your app to use Firebase Cloud Messaging as described on the [Firebase documentation](https://firebase.google.com/docs/cloud-messaging/android/client)
+You only need to setup the dependencies and _google-services.json_ file to your project source directory.
 
-#### Step 6
-Setup the following push notification data payload at Stream Dashboard:
-```json
-{
-  "message_id": "{{ message.id }}",
-  "channel_id": "{{ channel.id }}",
-  "channel_type": "{{ channel.type }}"
+#### Step 2
+Add the new artifact to the build gradle
+```groovy
+dependencies {
+    implementation "io.getstream:stream-chat-android-pushprovider-firebase:$stream_version"
 }
 ```
 
-#### Step 7
-Initialize _ChatClient_ with _NotificationConfig_:
-
+#### Step 3
+Add the `FirebasePushDeviceGenerator` to your NotificationConfig
 ```kotlin
-val notificationsConfig = NotificationConfig(
-    firebaseMessageIdKey = "message_id",
-    firebaseChannelIdKey = "channel_id",
-    firebaseChannelTypeKey = "channel_type",
+val notificationConfig = NotificationConfig(
+  pushDeviceGenerators = listOf(FirebasePushDeviceGenerator())
 )
-
 ChatClient.Builder("apiKey", context)
     .notifications(ChatNotificationHandler(context, notificationsConfig))
     .build()
 ```
 
+#### Implement Custom _FirebaseMessagingService_
+If you already are using FCM in your App, you can use your custom `FirebaseMessagingService` to receive push notifications from Stream
+
+```kotlin
+class CustomFirebaseMessagingService : FirebaseMessagingService() {
+
+    override fun onNewToken(token: String) {
+        // Update device's token on Stream backend
+        try {
+            ChatClient.setDevice(
+                Device(
+                    token = token,
+                    pushProvider = PushProvider.FIREBASE,
+                )
+            )
+        } catch (exception: IllegalStateException) {
+            // ChatClient was not initialized
+        }
+    }
+
+    override fun onMessageReceived(message: RemoteMessage) {
+        try {
+            // Check if the message from Firebase contains needed data and generate a PushMessage instance
+            val pushMessage = PushMessage(
+                channelId = data["channel_id"]!!,
+                messageId = data["message_id"]!!,
+                channelType = data["channel_type"]!!,
+            )
+            ChatClient.handlePushMessage(pushMessage)
+        } catch (exception: IllegalStateException) {
+            // ChatClient was not initialized
+        }
+    }
+}
+```
+
+:::note
+Make sure that CustomMessageFirebaseService's priority is higher than -1 to override the default service
+:::
+
+
+### Stream Huawei PushProvider implementation
+It is not ready yet, but we will implement it soon and release on the following releases. Be tuned
```

Comment:
`release on the following releases` I'm not sure if it is **in** or **on**.

Reply:
How about: `We are working on it currently and will release it shortly. Stay tuned.`